### PR TITLE
Project MongoDB week 14

### PR DESCRIPTION
**Pull request by NinaWald**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,46 @@
...
+## What you will learn
+✓ What MongoDB is
+✓ How to model data in Mongo using Mongoose
+✓ How to store secret information such as database passwords
+✓ How to return useful error messages from your API endpoints
+✓ How to fetch items from a Mongo database using Mongoose
+✓ How to seed large amounts of data to a database
 
 ## The problem
+I downloaded some CVSfile from Kaggle (Harry Potter info) and converted in into JSON (https://csvjson.com/)
```
Comment (Joanna-Philips) [inline README.md:18]: A tiny typo in CSV

File: server.js [hunk 2]
```diff
@@ -24,12 +31,210 @@ const app = express();
...
+*/
+
+/*
+In the code block below the schema definitions, there is a conditional block of code that checks for a RESET_DATABASE environment variable. 
+If this variable is present, 
+it will delete all data from the Characters and Spells collections in the MongoDB database, 
+and then insert all the characters and spells from the characters.json and spells.json files into the respective collections. 
+This is useful for resetting the database to a known state during development or testing.
+*/
+if (process.env.RESET_DATABASE) {
```
Comment (Joanna-Philips) [inline server.js:108]: Nice work seeding two datasets

File: server.js [hunk 3]
```diff
@@ -24,12 +31,210 @@ const app = express();
...
+  try {
+    res.json(spellsPotter);
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ message: "Server error" })
+  }
+});
+
+
+app.get("/characters/ID/:ID", async (req, res) => {
```
Comment (Joanna-Philips) [inline server.js:163]: I am not able to fetch character by ID. Not sure if the param entered is wrong.
This was my url:
`https://project-mongo-api-pb7rmnzmyq-lz.a.run.app/characters/ID/2`
It throws this error:
`"Cast to ObjectId failed for value \"2\" (type string) at path \"_id\" for model \"Characters\"`

File: server.js [hunk 4]
```diff
@@ -24,12 +31,210 @@ const app = express();
...
+        message: e
+      }
+    })
+  }
+});
+
+app.get("/characters/name/:name", async (req, res) => {
+  try {
+    const characters = await Characters.find({
+      "Character Name": { $regex: req.params.name, $options: "i" }
```
Comment (Joanna-Philips) [inline server.js:192]: Good job using regex and insensitizing case for the names!

**Review by Joanna-Philips — COMMENTED** [review]
Nina you have done a really good job with the backend part of this project. ⭐ 
And the fact that you have used datasets is great. 
I did notice an error while fetching and have mentioned it in the comments.

Overall. really good work! 🎉

File: server.js (hunk 3 above)
Comment (NinaWald) [reply server.js:163]: Ok great, I changed it now. The reason the user couldn't find the character by using ID/(ex 1 for Harry) is that the endpoint I had defined is looking for the MongoDB _id field, not the Character ID field. The findById method in Mongoose is designed to find documents by their unique _id field, not other fields in the document.
To allow users to search for a character by their Character ID, I needed to modify the endpoint to use the findOne method in Mongoose, which allows you to search by any field in the document.
With this modification, users will be able to search for a character by using their Character ID field, for example, https://project-mongo-api-pb7rmnzmyq-lz.a.run.app/characters/ID/1 (Harry Potter)

**Pull request closed without merging**